server/subscribe.go

```diff
@@ -647,50 +646,10 @@ func (p *Plugin) hasPermissionToManageSubscription(instanceID types.ID, userID,
 	return nil
 }
 
-func (p *Plugin) atomicModify(key string, modify func(initialValue []byte) ([]byte, error)) error {
-	readModify := func() ([]byte, []byte, error) {
-		initialBytes, appErr := p.API.KVGet(key)
-		if appErr != nil {
-			return nil, nil, errors.Wrap(appErr, "unable to read initial value")
-		}
-
-		modifiedBytes, err := modify(initialBytes)
-		if err != nil {
-			return nil, nil, errors.Wrap(err, "modification error")
-		}
-
-		return initialBytes, modifiedBytes, nil
-	}
-
-	var (
-		retryLimit     = 5
-		retryWait      = 30 * time.Millisecond
-		success        = false
-		currentAttempt = 0
-	)
-	for !success {
-		initialBytes, newValue, err := readModify()
-
-		if err != nil {
-			return err
-		}
-
-		var setError *model.AppError
-		success, setError = p.API.KVCompareAndSet(key, initialBytes, newValue)
-		if setError != nil {
-			return errors.Wrap(setError, "problem writing value")
-		}
-
-		if currentAttempt == 0 && bytes.Equal(initialBytes, newValue) {
-			return nil
-		}
-
-		currentAttempt++
-		if currentAttempt >= retryLimit {
-			return errors.New("reached write attempt limit")
-		}
-
-		time.Sleep(retryWait)
+func (p *Plugin) atomicModify(key string, modify func(initialValue []byte) (interface{}, error)) error {
+	err := p.client.KV.SetAtomicWithRetries(key, modify)
```

Comment:
Is `p.atomicModify` need any longer? Can we use `p.client.KV.SetAtomicWithRetries`?